### PR TITLE
Update README to add DialogStringsHolder #49

### PR DESCRIPTION
**Pull request by albinpoignot**
And this is the doc :)

(my markdown editor seems to have removed unneeded spaces as well)

### CODE REVIEW

**Comment by stkent** [comment]
For viewing non-whitespace-only changes: https://github.com/btkelly/gandalf/pull/50/files?w=1

File: README.md [hunk 1]
```diff
@@ -112,9 +112,49 @@ You must host a JSON file remotely and set the URL of this file in the the Ganda
 
 
 That's all that's needed to get Gandalf up and running using the basic settings.
- 
+
 If extending `GandalfActivity` doesn't work for you the `Gandalf` class can be used directly by calling `shallIPass(GandalfCallback callback)`. In this case make sure you respond to the callback methods and make a call to `gandalf.save(Alert alert)` and `gandalf.save(OptionalUpdate optionalUpdate)` if not using the `BootstrapDialogUtil` for your UI.
 
+#### Custom titles, buttons and messages
+
+By default, Gandalf provides default title and button text, and get the message to display to the user from the JSON file.
```
Comment (jsibbold) [inline README.md]: "and gets the message"

File: README.md [hunk 2]
```diff
@@ -112,9 +112,49 @@ You must host a JSON file remotely and set the URL of this file in the the Ganda
...
+
+However, you are able to use your own strings. To do so, you should use the `DialogStringHolder` class when installing Gandalf.
+
+1. If you do not provide a `DialogStringHolder` during installation, a default instance will be used.
+2. If you do not provide message strings in the `DialogStringHolder`, the message from the JSON file will be used.
+3. If you provide `DialogStringHolder` but do not set some field manually, default values will be used instead.
+4. You could either pass a `String` instance or a string resource id.
+
+**Remember**: you are not forced to set every string : default values will be used for unset string.
```
Comment (jsibbold) [inline README.md]: "default values will be used for all unset strings"

**Comment by jsibbold** [comment]
good additions to the docs, just a couple small grammar fixes

**Comment by albinpoignot** [comment]
Done. Thank you for the fixes and sorry for my english ^^

**Comment by btkelly** [comment]
Looks good, Approved

**Comment by btkelly** [comment]
Approved

**Pull request merged**